### PR TITLE
Improve Security on Form Actions

### PR DESCRIPTION
**Pull request by badbreze**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | 

This PR is intended to secure the Action of the Forms by cleaning up malicious codes from the provider URL as much as possible to avoid XSS

### CODE REVIEW

**Comment by mtangoo** [comment]
@badbreze can you please respond to the comment by @rob006?

**Comment by badbreze** [comment]
> How is this supposed to prevent XSS?

the filter prevents malicious code to be injected from the browser URL to the jQuery selector arguments.
in this case the Action of the form.

**Comment by rob006** [comment]
> the filter prevents malicious code to be injected from the browser URL to the jQuery selector arguments.
> in this case the Action of the form.

I that case jQuery should encode `action` attribute, which prevents XSS. XSS protection relies on proper encoding of content, not on stripping some chars from data.

**Comment by Webkadabra** [comment]
We need to start banning people, this is terrorism. Totally nonsense PR. Are you really still using Yii's client-side scripts, to the point where security is a concern? Really? Please, just waste your time elsewhere, preferably learning CS

**Comment by mtangoo** [comment]
@Webkadabra Learning discussing without name calling and intimidation is possible and encouraged in Yii community.
We encourage people here to make PR. We debate it based on merits and agree to merge or close based on merits.
Your comment is not with that spirit, in places you comment.

Please take note and correct that one.
Welcome to Yii community!

**Comment by mtangoo** [comment]
> The thing is that we're passing arguments to jQuery directly from the URL, an extremely vulnerable door to the application code, so the PR is (at least in my case) a way to try securing this possible flaw, but i let you decide if this can be accepted or not.

I get your point. tried to check and see how JQuery deals with encoding, I could not find decent docs. But if you want to propertly encode, may be you should look into native functions for that than simple  regex replacement? Something like [encodURI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI)?

**Comment by rob006** [comment]
This is not about encoding an URL, but about encoding a HTML attribute. I think we can safely assume that the most popular JS library can do such basic task. :)

**Comment by Webkadabra** [comment]
> @Webkadabra Learning discussing without name calling and intimidation is possible and encouraged in Yii community. We encourage people here to make PR. We debate it based on merits and agree to merge or close based on merits. Your comment is not with that spirit, in places you comment.
> 
> Please take note and correct that one. Welcome to Yii community!

please point to the name calling you're so offended by

**Comment by mtangoo** [comment]
> This is not about encoding an URL, but about encoding a HTML attribute. I think we can safely assume that the most popular JS library can do such basic task. :)

Let me have little sleep, lol :)

**Comment by Webkadabra** [comment]
> > please point to the name calling you're so offended by
> 
> "Totally nonsense PR" The rest can be read on your comment it is still there. You couldn't discuss PR without calling it nonsense? And there are other comments in other PR, that are offensive as well.

this is factually a nonsense PR, too bad you're taking coding personally. You shouldv'e been trying to get good at coding, but you mastered getting offended

**Comment by rob006** [comment]
@Webkadabra I suggest to you to pick one of many open [issues](https://github.com/yiisoft/yii2/issues), create a PR to fix it and show us all how good coding looks like. So far all your comments were extremity unconstructive and probably wasted more people's time and mental energy than PRs you were commenting.

**Comment by Webkadabra** [comment]
> @Webkadabra kindly take @rob006's advice and help us with your great skills iron bugs. It is welcome and appreciated. Bragging and discouraging people who took their time to make actual PR is discouraged and unwelcome.
> 
> Again take note that you need to be respectful towards others here, even if you think you are the genius in the room!

You tell me what to do, lie about name calling, then ask me to be respectful. Are you out of your mind? Who gave you the computer, did they not tell you to behave well?

**Comment by Webkadabra** [comment]
Dammit, soft-bodies boys not can't write code, don't understsand framework and scope, try to argue. Ok, I bet your consulting business is thriving. 

> @Webkadabra I suggest you to pick one of many open [issues](https://github.com/yiisoft/yii2/issues), create a PR to fix it and show us all how good coding looks like. So far all your comments were extremity unconstructive and probably wasted more people's time and mental energy than PRs you were commenting.

Oh sir with the advice, have you seen these braindead issues? A guy creates an infinite loop and claims it's Yii memory leak. Really, pick an issue, haha. The main issue is PR and issue spamming right now, with some villagers yelling god knows what

**Comment by Webkadabra** [comment]
> @Webkadabra I suggest you to pick one of many open [issues](https://github.com/yiisoft/yii2/issues), create a PR to fix it and show us all how good coding looks like. So far all your comments were extremity unconstructive and probably wasted more people's time and mental energy than PRs you were commenting.

Oh sir with the advice, have you seen these braindead issues? A guy creates an infinite loop and claims it's Yii memory leak. Really, pick an issue, haha. The main issue is PR and issue spamming right now, with some villagers yelling god knows what

**Comment by schmunk42** [comment]
@samdark @bizley Just block this guy, this is leading nowhere.

**Comment by Webkadabra** [comment]
> @samdark @bizley Just block this guy, this is leading nowhere.

Yes @samdark  please block mtangoo

**Comment by Webkadabra** [comment]
> @samdark @bizley Just block this guy, this is leading nowhere.

There are two pages full of sonsense issues, and this is one of them. I hope this leads to not accepting this PR, as well as many others

**Comment by samdark** [comment]
@schmunk42 done. @Webkadabra if you want to be un-blocked, please contact me directly and be prepared to correct the way you communicate.

File: framework/assets/yii.js [hunk 1]
```diff
@@ -223,7 +223,7 @@ window.yii = (function ($) {
                 }
             } else {
                 if (!isValidAction) {
-                    action = pub.getCurrentUrl();
+                    action = pub.getSafeUrl();
                 }
                 $form = $('<form/>', {method: method, action: action});
```
Comment (samdark) [inline framework/assets/yii.js:228]: Hmm... jQuery doesn't encode attributes?

**Pull request closed without merging**